### PR TITLE
solution

### PR DESCRIPTION
**Pull request by vladyslav-harkusha**
[DEMO LINK](https://vladyslav-harkusha.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by JHesher — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -3,29 +3,138 @@ import './App.scss';
...
+
+    this.setState({ todos: [...todosFromServer] });
+  }
+
+  componentDidUpdate = (_: Props, prevState: State) => {
+    if (prevState.selectedFilter !== this.state.selectedFilter) {
+      switch (this.state.selectedFilter) {
+        case 'active':
+          filterTodos(false)
+            .then(todos => {
```
Comment (Elena-brv) [inline src/App.tsx:35]: better use async/await

File: src/api/api.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+export const getTodos = (): Promise<Todo[]> => (
+  getData('todos')
+);
+
+export const getUser = (userId: number): Promise<User> => (
+  getData(`users/${userId}`)
+);
+
+export const filterTodos = (completed: boolean): Promise<Todo[]> => (
```
Comment (Elena-brv) [inline src/api/api.ts:17]:
```suggestion
export const getFilteredTodosByStatus = (completed: boolean): Promise<Todo[]> => (
```

File: src/components/CurrentUser/CurrentUser.tsx [hunk 3]
```diff
@@ -1,12 +1,79 @@
...
+};
+
+export class CurrentUser extends React.Component<Props, State> {
+  state: State = {
+    isUser: true,
+    user: {
+      id: 0,
+      name: '',
+      phone: '',
+      email: '',
```
Comment (Elena-brv) [inline src/components/CurrentUser/CurrentUser.tsx:22]: why do you need to write it here?

**Review by Elena-brv — CHANGES_REQUESTED** [review]
no very good to see it while the user info is loading
image

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**